login-utils/login.c

```diff
@@ -1310,20 +1306,11 @@ static void load_credentials(struct login_context *cxt) {
 		return;
 	}
 
-	dir = ul_path_opendir(pc, NULL);
-	if (!dir) {
-		syslog(LOG_WARNING, _("failed to open credentials directory"));
-		return;
-	}
-
-	while ((d = xreaddir(dir))) {
-		char str[32] = { 0 };
+	if (ul_path_read_buffer(pc, str, sizeof(str), "login.noauth") > 0
+	    && *str && strcmp(str, "yes") == 0)
+		cxt->noauth = 1;
 
```

Comment:
The function is designed to be in the future extended to support more credentials files. See https://github.com/util-linux/util-linux/pull/2120#issuecomment-1476047355

It would be probably better to remove this change from the patch (PR).

Reply:
To me it feels nicer to have a few direct file accesses than the current readdir loop.

Reply:
Hmm... I've thought originally about error messages for unsupported credentials, but it's probably overkill because it will fill system logs.

OK, @stoeckmann keep the change in your PR and remove the readdir(). We can change it later, after release we will see how people will use it ... (I have doubts about the attractivity of all the credential features, so maybe keeping it minimalistic is a good idea).